### DISCUSSION TITLE
MySQL max-retries to prevent boot failure

### DISCUSSION THREAD

**Discussion by Zolka42** [gerbera/gerbera#2390] [Ideas]
Hi All

I am trying to use gerbera on my home-server (FreeBSD) with MySQL database. The problem I am facing is that gerbera fails to start when the system boots up, because at the time gerbera gets started, MySQL is not ready.
I tried to find a solution, e.g.: delay the start of gerbera for a few seconds, buuut....
The real solution would be for gerbera to provide a config setting for MySQL, like: max-retries. This setting would prevent gerbera to fail instantly if it cannot access the database. It should wait like 1 second, then try again for maybe 10 times, and fail after that. With that, the users can have a more stable startup at boot time. I guess this issue is possible for other systems too.

What do you think?

**Comment by whyman** [comment]
Great idea

**Reply by KarlStraussberger** [reply]
On systems with systemd this is handled by 
```
After=mysql.target network-online.target
```
in the unit file. Systemd wait fo mysql to come up before starting gerbera.
